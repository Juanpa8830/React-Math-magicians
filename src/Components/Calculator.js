import { useState } from 'react';
import NumbButton from './numb-button';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
    display: '0',
  });

  const eventClick = (buttonName) => {
    let display = '';
    const { total, next, operation } = calculate(state, buttonName);

    if (operation !== null && operation !== undefined) {
      display = next !== null && next !== undefined ? next : total;
    } else if (next !== null && next !== undefined) {
      display = next;
    } else if (total !== null && total !== undefined) {
      display = total;
    }
    setState({
      total,
      next,
      operation,
      display,
    });
  };

  const { display } = state;

  const classChain = 'AC,mm,porcentaje,division,nueve,ocho,siete,seis,cinco,cuatro,tres,dos,uno,cero,decimal,multiplicar,restar,sumar,igual';
  const ClArr = classChain.split(',');
  const elementsChain = 'AC,+/-,%,รท,9,8,7,6,5,4,3,2,1,0,.,x,-,+,=';
  const elemArray = elementsChain.split(',');

  const btnList = elemArray.map((El, i) => {
    const btnkey = `btn${i}`;
    return (
      <NumbButton addClass={ClArr[i]} element={El} functionClick={eventClick} key={btnkey} />
    );
  });

  return (
    <div className="calc-section">
      <h2 className="calc-h2">Let&apos;s do some math!!</h2>
      <section className="calculator-container">
        <p className="display">{display}</p>
        {btnList}
      </section>
    </div>
  );
};

export default Calculator;
