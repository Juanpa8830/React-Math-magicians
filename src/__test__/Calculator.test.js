import React from 'react';
import { createRoot } from 'react-dom/client';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('renders without crashing', () => {
  const div = document.createElement('div');
  const root = createRoot(div);
  root.render(<Calculator />);
});

test('match Calculator snapshot', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
