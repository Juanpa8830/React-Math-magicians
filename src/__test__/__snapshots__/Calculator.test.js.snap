// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match Calculator snapshot 1`] = `
<div
  className="calc-section"
>
  <h2
    className="calc-h2"
  >
    Let's do some math!!
  </h2>
  <section
    className="calculator-container"
  >
    <p
      className="display"
    >
      0
    </p>
    <button
      className="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="mm"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="porcentaje"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="division"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="nueve"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="ocho"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="siete"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="seis"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="cinco"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="cuatro"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="tres"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="dos"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="uno"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="cero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="decimal"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="multiplicar"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="restar"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="sumar"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="igual"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </section>
</div>
`;
